docs/source/locking.rst

```diff
@@ -0,0 +1,80 @@
+Locking
+=======
+
```

Comment:
@buchi we should add a short introduction what locking is. I would like to write the documentation in a way that a developer without deep knowledge of zope/plone can understand the basics.